### PR TITLE
#4186 Fix caching, add log messages

### LINKED ISSUE #4186: Keptn on Keptn: Mark test as failed if a delivery fails
Right now there is no way of us knowing that an evaluation in our Meta Keptn fails, other than manually going into Keptn's Bridge:

image

The delivery is started here:
https://github.com/keptn/keptn/blob/0f24233122a216b6bf597731f8673f345b841e0c/.github/workflows/integration_tests.yml#L984-L1011

We should also verify that the delivery has succeeded, e.g., by taking the shkeptncontext of the delivery and querying for `${STAGE}.delivery.finished` events

# Acceptance Criteria

- [ ] Integration test pipeline fails if delivery of Keptn with Keptn fails
- [ ] Integration test pipeline fails if delivery of jmeter-service with Keptn fails
- [ ] Integration test pipeline fails if delivery of helm-service with Keptn fails

### PR DESCRIPTION
**Pull request by mowies**
**This PR**
* fixes the caching step for meta keptn delivery triggers
* adds some minor log output messages for better readability
* moves some environment variables from the `.ci_env` to the pipeline to optimize pipeline runtime

Fixes #4186

Signed-off-by: Moritz Wiesinger <moritz.wiesinger@dynatrace.com>

### CODE REVIEW

File: .github/workflows/integration_tests.yml [hunk 1]
```diff
@@ -1118,20 +1125,28 @@ jobs:
...
 
-      - name: Install Keptn CLI for META_KEPTN
-        id: install_keptn_cli
+      - name: Download Keptn CLI for META_KEPTN
+        id: download_keptn_cli
         if: steps.keptn-cli-cache.outputs.cache-hit != 'true'
         run: |
-          curl -sL https://get.keptn.sh | KEPTN_VERSION=${{ env.META_KEPTN_VERSION }} bash
+          curl -sL https://get.keptn.sh -o ${{ env.KEPTN_INSTALLER_PATH }}
+          chmod +x ${{ env.KEPTN_INSTALLER_PATH }}
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml:1145]: This is not what you want here.

This downloads the install script, but not the Keptn CLI binary.

Please use the following URL to download the CLI binary:
```
https://github.com/keptn/keptn/releases/download/${META_KEPTN_VERSION}/keptn-${META_KEPTN_VERSION}-linux-amd64.tar.gz
```

File: .github/workflows/integration_tests.yml [hunk 2]
```diff
@@ -1118,20 +1125,31 @@ jobs:
...
+      - name: Download Keptn CLI for META_KEPTN
+        id: download_keptn_cli
         if: steps.keptn-cli-cache.outputs.cache-hit != 'true'
         run: |
-          curl -sL https://get.keptn.sh | KEPTN_VERSION=${{ env.META_KEPTN_VERSION }} bash
+          curl -sL https://github.com/keptn/keptn/releases/download/${META_KEPTN_VERSION}/keptn-${META_KEPTN_VERSION}-linux-amd64.tar.gz -o ${{ env.KEPTN_ARCHIVE_PATH }}
+          tar -xvf ${{ env.KEPTN_ARCHIVE_PATH }}
+          mv ./keptn-${META_KEPTN_VERSION}-linux-amd64 keptn
+          chmod +x keptn
+          mv ./keptn /usr/local/bin/keptn
```
Comment (christian-kreuzberger-dtx) [inline .github/workflows/integration_tests.yml:1148]: This might require `sudo`

Comment (mowies) [reply .github/workflows/integration_tests.yml:258]: i will try it out and add it if there's no other way

Comment (mowies) [reply .github/workflows/integration_tests.yml]: it seems that no `sudo` is needed :)
https://github.com/keptn/keptn/runs/2894630082?check_suite_focus=true#step:5:4

File: .github/workflows/integration_tests.yml (hunk 1 above)
Comment (mowies) [reply .github/workflows/integration_tests.yml]: good catch! done ✅

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM

**Pull request merged**